src/test/java/AlgorithmHandler/algorithms/FileHandleAlgorithm.java

```diff
@@ -0,0 +1,18 @@
+package AlgorithmHandler.algorithms;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+
+public class FileHandleAlgorithm {
+    public FileOutputStream foo(String localFile) {
+        try {
+            File file = new File(localFile);
+            file.createNewFile();
+            file.delete();
+            return new FileOutputStream(file);
```

Comment:
you're deleting a file and then returning an output stream to that file...? that doesn't seem right, and also this function would need the caller to close the output stream (not sure i see where it's used anyway)

Reply:
This is entirely a "fail gracefully with odd return type" test, what the algorithm does isn't important, it could just be `return FileOutputStream(new File('.'))`

Reply:
If you have a recommended junit testing style or doc I'd love to read it, this is my first foray into TTD

Reply:
it's just a matter of having output streams that don't get properly cleaned up anywhere is a scary thing for me, even if it's just in test code.  and what do you mean `odd return type`? is that really a valid scenario to even test?

Reply:
The point here is to have a failure message to the developer that articulates a problem around a particular return type, the point here is we want to find failure cases that novices might run into, and make sure that we return valuable and useful error messages when that happens. a FileOutputStream is obviously not a serializable type, and will error out.